api/credentials-config-service.yaml

```diff
@@ -0,0 +1,282 @@
+                format: uri
+                example: /service/packet-delivery-service
+              description: Location of the created service
+        '400':
+          description: Invalid service provided
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '409':
+          description: Service with the given id already exists.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+    get:
+      tags:
+        - service
+      parameters:
+        - $ref: '#/components/parameters/PageSize'
+        - $ref: '#/components/parameters/Offset'
+      operationId: getServices
+      summary: Return all services
+      description: Return all services configured, with pagination.
+      responses:
+        '200':
+          description: The service config.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Services'
+        '400':
+          description: Invalid query parameters provided
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+
+  /service/{id}:
+    get:
+      tags:
+        - service
+      parameters:
+        - $ref: '#/components/parameters/Id'
+      operationId: getService
+      summary: Return the full service config by ID
+      description: The service configuration, including all credentials and their trust anchors will be returned.
+      responses:
+        '200':
+          description: The service config
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Service'
+        '404':
+          description: No such service exists.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+    delete:
+      tags:
+        - service
+      parameters:
+        - $ref: '#/components/parameters/Id'
+      operationId: deleteServiceById
+      description: Delete a single service(and all its configurations) with the given id.
+      summary: Delete the service
+      responses:
+        '204':
+          description: Successfully deleted
+        '404':
+          description: No such service exists.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+    put:
+      tags:
+        - service
+      parameters:
+        - $ref: '#/components/parameters/Id'
+      summary: Update a single service
+      description: Updates a single service by fully overriding it.
+      operationId: updateService
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Service'
+      responses:
```

Comment:
Do we need a 409 in case we change the serviceId in the payload and the target id exists?

Reply:
I would not allow this and throw a 400 in case path-id != payload-id